utils/hct/hctdb.py

```diff
@@ -2582,6 +2582,7 @@ def build_valrules(self):
         self.add_valrule("Instr.BarrierModeNoMemory", "sync must include some form of memory barrier - _u (UAV) and/or _g (Thread Group Shared Memory).  Only _t (thread group sync) is optional.")
         self.add_valrule("Instr.BarrierModeForNonCS", "sync in a non-Compute/Amplification/Mesh Shader must only sync UAV (sync_uglobal).")
         self.add_valrule("Instr.WriteMaskForTypedUAVStore", "store on typed uav must write to all four components of the UAV.")
+        self.add_valrule("Instr.WriteMaskGapForUAV", "UAV write mask must be contiguous and without gaps: .x, .xy, .xyz, or .xyzw.")
```

Comment:
I suppose another characteristic that is checked here is that it starts at x. The explicit examples make that clear though.